sklearn/metrics/pairwise.py

```diff
@@ -77,6 +77,7 @@ def check_pairwise_arrays(
     dtype=None,
     accept_sparse="csr",
     force_all_finite=True,
+    only_check_num_samples=False,
```

Comment:
this API to me is confusing. Why not allow strings and objects as dtype, and have another parameter called `ensure_2d` to skip 2d check explicitly?

Reply:
replaced `only_check_num_samples=False` with `ensure_2d=True`